flake.nix

```diff
@@ -0,0 +1,88 @@
+{
+  description = "way-displays: Auto Manage Your Wayland Displays";
+
+  # nixpkgs and flake-parts are in the flake registry, so we don't have to specify their input URLs
+  # https://github.com/NixOS/flake-registry/blob/master/flake-registry.json
+  inputs = {
+    pre-commit-hooks-nix = {
+      url = "github:cachix/pre-commit-hooks.nix";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs = inputs@{ nixpkgs, flake-parts, ... }:
+    # flake-parts modularizes flake construction
+    # https://flake.parts/
+    flake-parts.lib.mkFlake { inherit inputs; } {
+      imports = [
+        inputs.pre-commit-hooks-nix.flakeModule
+      ];
+
+      # It is necessary to explicitly list supported systems. Add more as needed from
+      # https://github.com/NixOS/nixpkgs/blob/master/lib/systems/flake-systems.nix
+      systems = [ "x86_64-linux" "aarch64-linux" ];
+      perSystem = { config, self', inputs', pkgs, system, ... }: {
+        # stdenv analyzes the Makefile etc and tries to "do the right thing" without much config
+        # https://static.domenkozar.com/nixpkgs-manual-sphinx-exp/stdenv.xml.html
+        # This is where the magic happens:
+        # https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/setup.sh
+
+        # Use clangStdenv (1.2GB closure) instead of regular gcc stdenv (322MB closure).
```

Comment:
Yes please. gcc defaults are only present for the package builds.

I've noted clang preference in CONTRIBUTING